ultrasonic/src/main/kotlin/org/moire/ultrasonic/fragment/EditServerFragment.kt

```diff
@@ -330,17 +338,65 @@ class EditServerFragment : Fragment(), OnBackPressedHandler {
                 pingResponse = subsonicApiClient.api.ping().execute()
                 ApiCallResponseChecker.checkResponseSuccessful(pingResponse)
 
+                updateProgress(String.format(progressString, "⌛", "⌛", "⌛", "⌛"))
```

Comment:
There is quite a bit of code duplication here. Maybe put the handling of which string to display into the UpdateProgress() function. And does this class ha e a viewmodel? If yes the actual connectio. Checking should go there...

Reply:
Updated
It's a bit cleaner now